Sources/Datadog/Core/Persistence/FileReader.swift

```diff
@@ -43,7 +42,7 @@ internal final class FileReader {
         if let file = orchestrator.getReadableFile(excludingFilesNamed: Set(filesRead.map { $0.name })) {
             do {
                 let fileData = try file.read()
-                let batchData = openingBracketData + fileData + closingBracketData
+                let batchData = dataFormat.prefixData + fileData + dataFormat.suffixData
```

Comment:
how about turning `DataFormat` into `DataFormatter` and doing `let batchData = formatter.finalize(fileData)` instead of making `FileReader` know the implementation details?

i'm not requesting this change 
because i see `try write(dataFormat.separatorData)` below may not be that easy 🤔

Reply:
Honestly, I find:
```swift
let batchData = dataFormat.prefixData + fileData + dataFormat.suffixData
```
more readable and straightforward than
```swift
let batchData = formatter.finalize(fileData)
```
no?

Reply:
it was more about taking the responsibility of formatting data out of `FileWriter` and giving it to `DataFormatter`

IMO either way is fine 👍